lib/widgets/topic_list.dart

```diff
@@ -0,0 +1,333 @@
+
+  final int streamId;
+  final _TopicItemData data;
+
+  @override
+  Widget build(BuildContext context) {
+    final _TopicItemData(
+      :topic, :unreadCount, :hasMention, :maxId) = data;
+
+    final store = PerAccountStoreWidget.of(context);
+
+    final designVariables = DesignVariables.of(context);
+    final isTopicVisibleInStream = store.isTopicVisibleInStream(streamId, topic);
+    final visibilityIcon = iconDataForTopicVisibilityPolicy(
+      store.topicVisibilityPolicy(streamId, topic));
+
+    final trailingWidgets = [
+      if (hasMention) const _IconMarker(icon: ZulipIcons.at_sign),
+      if (visibilityIcon != null) _IconMarker(icon: visibilityIcon),
+      if (unreadCount > 0) _UnreadCountBadge(count: unreadCount),
+    ];
+
+    return Material(
+      color: designVariables.bgMessageRegular,
+      child: InkWell(
+        onTap: () {
+          final narrow = TopicNarrow(streamId, topic);
+          Navigator.push(context,
+            MessageListPage.buildRoute(context: context, narrow: narrow));
+        },
+        onLongPress: () => showTopicActionSheet(context,
+          channelId: streamId,
+          topic: topic,
+          someMessageIdInTopic: maxId),
+        splashFactory: NoSplash.splashFactory,
+        child: Padding(padding: EdgeInsetsDirectional.fromSTEB(6, 8, 12, 8),
+          child: Row(
+            spacing: 8,
+            // In the Figma design, the text and icons on the topic item row
+            // are aligned to the start on the cross axis
+            // (i.e., `align-items: flex-start`).  The icons are padded down
+            // 2px relative to the start, to visibly sit on the baseline.
+            // To account for scaled text, we align everything on the row
+            // to [CrossAxisAlignment.center] instead ([Row]'s default),
+            // like we do for the topic items on the inbox page.
+            // CZO discussion:
+            //   https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/topic.20list.20item.20alignment/near/2173252
+            children: [
+              // A null [Icon.icon] makes a blank space.
+              _IconMarker(icon: topic.isResolved ? ZulipIcons.check : null),
+              Expanded(child: Opacity(
+                opacity: isTopicVisibleInStream ? 1 : 0.5,
+                child: Text(
+                  style: TextStyle(
+                    fontSize: 17,
+                    height: 20 / 17,
+                    fontStyle: topic.displayName == null ? FontStyle.italic : null,
+                    color: designVariables.textMessage,
+                  ),
+                  topic.unresolve().displayName ?? store.realmEmptyTopicDisplayName))),
+              Opacity(opacity: isTopicVisibleInStream ? 1 : 0.5, child: Row(
+                spacing: 4,
+                children: [
+                  ...trailingWidgets,
+                  if (trailingWidgets.isEmpty)
+                    const SizedBox(width: 53),
```

Comment:
I see this is what the Figma is doing, but it feels odd: it doesn't create a consistent end margin for the topic text, because the gap will be less than 53px when there's just one or two `trailingWidgets`, right? How about just removing this and allowing a longer line for the topic text.